malicious-site-protection/malicious-site-protection-impl/src/main/kotlin/com/duckduckgo/malicioussiteprotection/impl/domain/RealMaliciousSiteProtection.kt

```diff
@@ -78,28 +81,31 @@ class RealMaliciousSiteProtection @Inject constructor(
             return ConfirmedResult(Safe)
         }
 
-        val hash = messageDigest
-            .digest(hostname.toByteArray(Charsets.UTF_8))
-            .joinToString("") { "%02x".format(it) }
+        val hash = generateHash(hostname)
         val hashPrefix = hash.substring(0, 8)
 
-        if (!maliciousSiteRepository.containsHashPrefix(hashPrefix)) {
-            timber.d("should not block (no hash) $hashPrefix,  $canonicalUri")
-            return ConfirmedResult(Safe)
-        }
-        maliciousSiteRepository.getFilters(hash)?.forEach { filterSet ->
-            filterSet.filters.firstOrNull {
-                Pattern.compile(it.regex).matcher(canonicalUri.toString()).find()
-            }?.let {
-                timber.d("should block $canonicalUri")
-                return ConfirmedResult(Malicious(filterSet.feed))
+        try {
+            if (!maliciousSiteRepository.containsHashPrefix(hashPrefix)) {
+                timber.d("should not block (no hash) $hashPrefix,  $canonicalUriString")
+                return ConfirmedResult(Safe)
+            }
+            maliciousSiteRepository.getFilters(hash)?.let { filterSet ->
+                filterSet.filters.let {
+                    if (Pattern.compile(it.regex).matcher(canonicalUriString).find()) {
+                        timber.d("should block $canonicalUriString")
+                        return ConfirmedResult(Malicious(filterSet.feed))
+                    }
+                }
             }
+        } catch (e: WriteInProgressException) {
+            timber.d("Write in progress, ignoring")
+            return ConfirmedResult(Ignored)
         }
         appCoroutineScope.launch(dispatchers.io()) {
             try {
                 val result = when (val matches = maliciousSiteRepository.matches(hashPrefix.substring(0, 4))) {
                     is Result -> matches.matches.firstOrNull { match ->
-                        Pattern.compile(match.regex).matcher(url.toString()).find() &&
+                        Pattern.compile(match.regex).matcher(canonicalUriString).find() &&
```

Comment:
previously this was `url.toString()` now it's `canonicalUri.toString()`, is this correct? just checking